runtime/caml/mlvalues.h

```diff
@@ -148,16 +151,8 @@ bits  63        (64-P) (63-P)        10 9     8 7   0
 #define Op_hp(hp) ((value *) Val_hp (hp))
 #define Bp_hp(hp) ((char *) Val_hp (hp))
 
-#define Num_tags (1 << 8)
-#ifdef ARCH_SIXTYFOUR
-#ifdef WITH_PROFINFO
-#define Max_wosize (((intnat)1 << (54-PROFINFO_WIDTH)) - 1)
-#else
-#define Max_wosize (((intnat)1 << 54) - 1)
-#endif
-#else
-#define Max_wosize ((1 << 22) - 1)
-#endif /* ARCH_SIXTYFOUR */
+#define Num_tags (HEADER_TAG_MASK + 1)
+#define Max_wosize HEADER_WOSIZE_MASK
```

Comment:
I would prefer to have them defined in terms of `HEADER_TAG_BITS` and `HEADER_WOSIZE_BITS`, as I believe this makes more sense.